test/se/vidstige/jadb/test/fakes/FakeAdbServer.java

```diff
@@ -173,6 +193,32 @@ public void verifyExpectations() {
             org.junit.Assert.assertEquals(0, listExpectations.size());
         }
 
+        private static class TcpIpException implements ExpectationBuilder {
+
+            private String port;
+
+            public TcpIpException(final String port) {
+                this.port = port;
+            }
+
+            public boolean matches(String cmd) {
+                return cmd.equals(port);
+            }
+
+            @Override
+            public void failWith(String message) {
+
+            }
+
+            @Override
+            public void withContent(byte[] content) {
+
+            }
+
+            @Override
+            public void withContent(String content) {
+            }
+        }
```

Comment:
Lots of unused methods here. Lets not inherit from `ExpectationBuilder` just make a brand new class called `TcpIpExpectation` (note that this is an _expectation_, not an _exception_). Then you can remove all the empty functions and just keep the "matches" and the constructor. In fact, at this point, you can just have a list of `int`, each one being a port number where a `enableTcpIp` is expected. 🤷‍♂️ Less code, more win. :-)

Reply:
Great suggestion 👍